src/huggingface_hub/hf_api.py

```diff
@@ -972,56 +1004,103 @@ def model_info(
+            f"{self.endpoint}/api/datasets/{repo_id}"
+            if revision is None
+            else f"{self.endpoint}/api/datasets/{repo_id}/revision/{revision}"
+        )
+        headers = {"authorization": f"Bearer {token}"} if token is not None else None
+        r = requests.get(path, headers=headers, timeout=timeout)
+        r.raise_for_status()
+        d = r.json()
+        return DatasetInfo(**d)
 
-    def dataset_info(
+    def space_info(
         self,
         repo_id: str,
         revision: Optional[str] = None,
         token: Optional[str] = None,
         timeout: Optional[float] = None,
-    ) -> DatasetInfo:
+    ) -> SpaceInfo:
         """
-        Get info on one specific dataset on huggingface.co
+        Get info on one specific Space on huggingface.co
 
-        Dataset can be private if you pass an acceptable token.
+        Space can be private if you pass an acceptable token.
         """
         if token is None:
             token = HfFolder.get_token()
 
         path = (
-            f"{self.endpoint}/api/datasets/{repo_id}"
+            f"{self.endpoint}/api/spaces/{repo_id}"
             if revision is None
-            else f"{self.endpoint}/api/datasets/{repo_id}/revision/{revision}"
+            else f"{self.endpoint}/api/spaces/{repo_id}/revision/{revision}"
         )
         headers = {"authorization": f"Bearer {token}"} if token is not None else None
-        params = {"full": "true"}
-        r = requests.get(path, headers=headers, params=params, timeout=timeout)
+        r = requests.get(path, headers=headers, timeout=timeout)
         r.raise_for_status()
         d = r.json()
-        return DatasetInfo(**d)
+        return SpaceInfo(**d)
+
+    def repo_info(
+        self,
+        repo_id: str,
+        revision: Optional[str] = None,
+        repo_type: Optional[str] = None,
+        token: Optional[str] = None,
+        timeout: Optional[float] = None,
+    ) -> Union[ModelInfo, DatasetInfo, SpaceInfo]:
+        """
+        Get the info object for a given repo of a given type.
+        """
+        if repo_type is None or repo_type == "model":
+            return self.model_info(
+                repo_id, revision=revision, token=token, timeout=timeout
+            )
+        elif repo_type == "dataset":
+            return self.dataset_info(
+                repo_id, revision=revision, token=token, timeout=timeout
+            )
+        elif repo_type == "space":
+            return self.space_info(
+                repo_id, revision=revision, token=token, timeout=timeout
+            )
+        else:
+            raise ValueError("Unsupported repo type.")
+
+    def list_repo_files(
+        self,
+        repo_id: str,
+        revision: Optional[str] = None,
+        repo_type: Optional[str] = None,
+        token: Optional[str] = None,
+        timeout: Optional[float] = None,
+    ) -> List[str]:
+        """
+        Get the list of files in a given repo.
+        """
```

Comment:
same here with documentation.

Reply:
Since this was just moved around, this is handled by this PR already: https://github.com/huggingface/huggingface_hub/pull/782/files#diff-71a3e23532e1355e79edd51b351bd514f0aa55b946e4c530b0c95f647773ac5dR1097